src/main/java/burp/BurpExtender.java

```diff
@@ -50,23 +67,16 @@ public void processHttpMessage(int toolFlag, boolean messageIsRequest, IHttpRequ
         var httpService = messageInfo.getHttpService();
         var req = this.helpers.analyzeRequest(messageInfo.getRequest());
 
-        var transportConfig = new TransportConfig();
+        var transportConfig = new RequestConfig();
         transportConfig.Host = httpService.getHost();
         transportConfig.Scheme = httpService.getProtocol();
-        transportConfig.Fingerprint = this.settings.getFingerprint();
-        transportConfig.HexClientHello = this.settings.getHexClientHello();
-        transportConfig.HttpTimeout = this.settings.getHttpTimeout();
-        transportConfig.HttpKeepAliveInterval = this.settings.getHttpKeepAliveInterval();
-        transportConfig.IdleConnTimeout = this.settings.getIdleConnTimeout();
-        transportConfig.TlsHandshakeTimeout = this.settings.getTlsHandshakeTimeout();
```

Comment:
These were set here so you could change the transport config from burp settings and have them applied by the next request. This is no longer possible now that you have moved these within the thread above right? This was a nice feature though, do you think it's possible to adjust your implementation to support this again?

Reply:
it seems to me that it is not necessary, because now the settings are saved immediately after pressing the button, and you do not have to wait for a new query to apply them. Do you think it makes sense to support two ways of transferring settings to Go?

Reply:
Oh ok, I missed that then. 

>  because now the settings are saved immediately after pressing the button, and you do not have to wait for a new query to apply them.

This is good, thanks :+1: 

You have tested this right? If it works as described that's perfect so you may resolve this conversation.

Reply:
yes, i have tested it